### ISSUE TITLE
How to reduce time of Host-side TensorFlow operations(_Send, _HostSend) in GPU reference

### ISSUE THREAD

**Issue by AsherXian903** [tensorflow/tensorflow#51983]
Dear all,
       When I do a simple model refence on GPU, it seems that the kernel launch time is much more significant than compute time(kernel launch nearly 90% to overall). Accoring to the profiler result, the Host-side TensorFlow operations such as _Send and _HostSend type  seems to be the problem. Is there any solution to this problem?

Env: tensorflow2.2.0 + cuda10.1

Thanks,
Asher

PS:
overview_page:
![图片](https://user-images.githubusercontent.com/51115601/133186394-887ad5d5-f0cc-40e8-818d-e4b8e8b74b1b.png)
tensorflow_stats:
![图片](https://user-images.githubusercontent.com/51115601/133186502-0679d674-fb30-4574-b526-a4c879ce778a.png)

**Comment by mohantym** [comment]
Hi @AsherXian903 ,Could you please share a stand alone code to reproduce this issue as it will help expedite the issue!

**Comment by AsherXian903** [comment]
Here is the model:
[saved_model.tar.gz](https://github.com/tensorflow/tensorflow/files/7166175/saved_model.tar.gz)

Here is the code:
BATCH_SIZE = 10
LOOP_CNT = 10
def predict():
    print("predict")
    with tf.compat.v1.Session(config=tf.compat.v1.ConfigProto(gpu_options = tf.compat.v1.GPUOptions(per_process_gpu_memory_fraction=0.4))) as sess:
      # First load the SavedModel into the session
      model_dir = "./model_dir"
      tf.compat.v1.saved_model.loader.load(
          sess, [tf.saved_model.SERVING],
          model_dir)
      op_1 = sess.graph.get_tensor_by_name('user/logits/BiasAdd:0')
      op_2 = sess.graph.get_tensor_by_name('bias:0')
      input_schema = ["interest_category_all:0", "location_code2:0", "from:0", "os_version:0", "gender:0", "wm:0", "login_freq_num:0", "os_br
and:0", "platform_type:0", "interest_word_all:0", "lift_state_all:0", "age:0"]
      empty_list_10 = ["", "", "", "", "", "", "", "", "", ""]
      input_data = [2*empty_list_10, "30804", "10B5395010", "92210000", "402", "20005_0002", "30", "90103000", "android", ["美食", "运动", ">
美食博主", "吃", "健身", "运动健身", "健身运动", "健身达人", "食", "生活", "减肥", "好吃的", "萌宠", "料理", "美女", "衣服", "宠物", "动漫",
"吃的", "瘦身", "食物", "吃喝玩乐", "动物", "网红", "厨房"], empty_list_10, "1025"]
      input_tensor = {}
      for i in range(0, len(input_schema)):
        input_tensor[sess.graph.get_tensor_by_name(input_schema[i])] = []
        for j in range(0, BATCH_SIZE):
          input_tensor[sess.graph.get_tensor_by_name(input_schema[i])].append(np.reshape(input_data[i], (-1)).tolist())
      print("start local predict")
      t0 = time.time()
      tf.profiler.experimental.start('logdir')
      for i in range(0, LOOP_CNT):
        ret = sess.run([op_1, op_2], feed_dict=input_tensor)
      tf.profiler.experimental.stop()
      print(time.time() - t0, "seconds process time")
      print("end local predict")

**Comment by mohantym** [comment]
Hi @AsherXian903 ,Could you please check again with TF 2.5/2.6  ? please share a stand alone code for  model.fit() operation  tto check above performance parameters  .I checked the above code though, It ran fine in TF 2.6 in colab . providing [Gist ](https://colab.research.google.com/gist/mohantym/b9af2378dd1de2a65c06fa488a529fc7/github_51983.ipynb)for reference
